### PR TITLE
Fixes bug with display message logic due to deprecating GenericSource…

### PR DESCRIPTION
**Pull request by preetapan**
…. Also added more test cases to cover a bunch more edge cases

### CODE REVIEW

File: nomad/structs/structs.go [hunk 1]
```diff
@@ -3963,7 +3963,7 @@ func (event *TaskEvent) PopulateEventDisplayMessage() {
 	case TaskLeaderDead:
 		desc = "Leader Task in Group dead"
 	default:
-		desc = ""
+		desc = event.Message
```
Comment (chelseakomlo) [inline nomad/structs/structs.go]: Will event or message ever be nil? If the message is empty, should there be some generic output?

Comment (preetapan) [reply nomad/structs/structs.go]: Early return in line 3855 handles event being nil. If the message is empty the existing behavior leaves it as is, that's fine for now since the generic output can't add any more new information to the user.

File: nomad/structs/structs_test.go [hunk 2]
```diff
@@ -2433,24 +2434,47 @@ func TestACLPolicySetHash(t *testing.T) {
 }
 
 func TestTaskEventPopulate(t *testing.T) {
+	prepopulatedEvent := NewTaskEvent(TaskSetup)
+	prepopulatedEvent.DisplayMessage = "Hola"
 	testcases := []struct {
 		event       *TaskEvent
 		expectedMsg string
 	}{
 		{nil, ""},
```
Comment (preetapan) [inline nomad/structs/structs_test.go]: this is the test case for event being nil.
